src/formatDate.js

```diff
@@ -50,7 +50,55 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const separator = toFormat[3];
+  const arrDate = date.split(oldSeparator);
+  let day;
+  let month;
+  let year;
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'DD') {
+      day = arrDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = arrDate[i];
+    }
+
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = arrDate[i];
+    }
+  }
+
+  if (toFormat.includes('YY') && year.length > 2) {
+    year = year.slice(-2);
+  }
+
+  if (toFormat.includes('YYYY') && year.length < 4) {
+    if (year < 30) {
+      year = '20' + year;
+    } else {
+      year = '19' + year;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === 'DD') {
+      result[i] = day;
+    }
+
+    if (toFormat[i] === 'MM') {
+      result[i] = month;
+    }
+
+    if (toFormat[i] === 'YY' || toFormat[i] === 'YYYY') {
+      result[i] = year;
+    }
+  }
+
+  return result.join(separator);
```

Comment:
I've came up with that solution, you can check if that also works fine 😉 
```suggestion
  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
  const [fromYear, fromMonth, fromDay] = dateParts;

  let toYear = "";
  let toMonth = "";
  let toDay = "";

  for (let i = 0; i < toFormat.length; i++) {
    if (toFormat[i].includes("YY")) {
      // Convert year from 4 digits to 2 digits (YYYY to YY)
      toYear = fromYear.slice(-2);
    } else if (toFormat[i].includes("YYYY")) {
      // Convert year from 2 digits to 4 digits (YY to YYYY)
      const prefix = Number(fromYear.slice(0, 2)) < 30 ? "20" : "19";
      toYear = prefix + fromYear;
    } else if (toFormat[i].includes("MM")) {
      toMonth = fromMonth;
    } else if (toFormat[i].includes("DD")) {
      toDay = fromDay;
    }
  }

  const formattedDate = [toYear, toMonth, toDay].join(toFormat[toFormat.length - 1]);

  return formattedDate;

```

Reply:
Checked it. https://prnt.sc/8B9YxlskI2pz
Something is wrong with extending year length and day/year placement.

Reply:
sure, no problem 😄